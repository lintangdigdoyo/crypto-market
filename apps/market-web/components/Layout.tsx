import Head from "next/head";

interface LayoutProps {
  children: React.ReactNode;
}

const Layout = ({ children }: LayoutProps) => {
  return (
    <>
      <Head>
        <title>Crypto Market</title>
      </Head>
      <main className="max-w-7xl mx-auto my-2">{children}</main>
    </>
  );
};

export default Layout;
