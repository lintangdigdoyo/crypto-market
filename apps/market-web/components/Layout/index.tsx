import Head from "next/head";

interface LayoutProps {
  children: React.ReactNode;
}

const Layout = ({ children }: LayoutProps) => {
  return (
    <>
      <Head>
        <title>Crypto Market</title>
        <meta name="description" content="Harga Crypto dalam Rupiah Hari Ini" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <main className="mx-auto my-6 max-w-7xl">{children}</main>
    </>
  );
};

export default Layout;
