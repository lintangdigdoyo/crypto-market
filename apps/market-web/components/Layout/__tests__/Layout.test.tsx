import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

import Layout from "..";

describe("Layout component", () => {
  it("should renders layout correctly", () => {
    const props = {
      children: <>children</>,
    };
    const component = renderer.create(<Layout {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
