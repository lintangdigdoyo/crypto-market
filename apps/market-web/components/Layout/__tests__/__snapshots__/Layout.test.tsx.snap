// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should renders layout correctly 1`] = `
<main
  className="mx-auto my-6 max-w-7xl"
>
  children
</main>
`;
